components/financial_assistance/app/domain/financial_assistance/operations/transfers/medicaid_gateway/automatic_submission.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+require 'dry/monads/do'
+
+module FinancialAssistance
+  module Operations
+    module Transfers
+      module MedicaidGateway
+        # This class is used to automatically submit the application to Medicaid Gateway for determination.
+        class AutomaticSubmission
+          include Dry::Monads[:result, :do]
+          def call(application)
+            # Additional steps that may be needed:
+            # populate default values for application
+            # apply more validations (such as ones that are only enforced in the UI)
+            result = yield request_determination(application)
+            Success(result)
+          end
+
+          def request_determination(application)
+            Rails.logger.info "Submitting application #{application.id} to Medicaid Gateway for determination"
+
+            result = FinancialAssistance::Operations::Applications::MedicaidGateway::RequestEligibilityDetermination.new.call(application_id: application.id)
+            if result.failure?
+              Rails.logger.error "Failed automatic submission for application #{application.id} due to #{result.failure.inspect}"
+              return result
```

Comment:
Are we intentionally avoiding calling Failure here to allow this to fail silently?

Reply:
This will actually return the `Failure` monad object in the `result` variable and cause the `yield` on line 17 to halt the operation, so the `Success` on line 18 is never reached.  Calling `result.failure` would unwrap the monad and return its contents.

To the broader point though, the `AccountTransferIn` operation where this operation is called does not do anything with the `Failure` object that is returned.  That is intentional so that the account can still be successfully transferred into Enroll as a draft app even if automatic submission fails.

Reply:
Awesome! Thanks for the thorough explanation.